src/Azure.Functions.Cli/Actions/KubernetesActions/KubernetesDeployAction.cs

```diff
@@ -148,12 +152,20 @@ public override async Task RunAsync()
                     // TODO: Join this with the build above so that it can run in parallel to the Kubernetes object create
                     tasks.Add(DockerHelpers.DockerPush(resolvedImageName, false));
                 }
+                if (WriteConfigs)
+                {
+                    var yaml = KubernetesHelper.SerializeResources(resources, OutputSerializationOptions.Yaml);
+                    await File.WriteAllTextAsync(ConfigFile, yaml);
+                    Console.Write($"Configuration written to {ConfigFile}");
+                    return;
+                }
+                
```

Comment:
Should this be an `else if` on the outer if/else? 

```
if (DryRun)
{
    ....
}
else if (WriteConfigs)
{
    ....
}
else
{
    ...
}
```

Otherwise we're not awaiting the createNamespace or the docker build tasks which will cause them to start but finish randomly.

Reply:
Cleaned up the logic here a bunch, please take another look.